src/scm_diff_editor.rs

```diff
@@ -1091,8 +1091,12 @@ mod tests {
             Ok(self
                 .files
                 .keys()
-                .filter(|path| path.starts_with(left) || path.starts_with(right))
-                .cloned()
+                .filter_map(|path| {
```

Comment:
suggestion: I think to make this clearest logically, we want to actually generate two sets of paths and union them together, one per side, something like this:

```rust
let left = self.files.keys().filter_map(|path| path.strip_prefix(left));
let right = self.files.keys().filter_map(|path| path.strip_prefix(right));
Ok(left.chain(right).map(|path| path.to_path_buf()).collect()) // implicitly deduplicates by collecting into `BTreeSet`
```

Reply:
Thanks. This is much clearer than what I had originally.